src/versions.jl

```diff
@@ -262,7 +262,12 @@ subset of the pool of available versions, this function computes a `VersionSpec`
 includes all versions in `subset` and none of the versions in its complement.
 """
 function compress_versions(pool::Vector{VersionNumber}, subset::Vector{VersionNumber})
-    subset = sort(subset) # must copy, we mutate this
+    # Explicitly drop prerelease/build numbers, as those can confuse this.
+    # TODO: Rewrite all this to use VersionNumbers instead of VersionBounds
```

Comment:
As a heads up, in #1357 I am moving this stuff into `bin` since it is unused and untested in Pkg.

Reply:
Eeh, actually it is tested...

Reply:
Oh  because the relevant PR added tests... Anyway, this seems like a registry specific thing (there is no need to compress anything from Pkgs side so it seems it should live in Registrator.jl (which we will refactor out a RegisterTools.jl from).